### DISCUSSION TITLE
BitOps Meeting (07 Dec 2022): BitOps in Github Actions, CLI Ops Repo Generator, v2.3.0 Planning, Local plugin install

### DISCUSSION THREAD

**Discussion by arm4b** [bitovi/bitops#370] [Announcements]
We invite everyone (developers, adopters, or just curious people) to join the bi-weekly BitOps Community `1 hour` Meeting on Wednesday, `07 Dec 2022, 12:00 PM US East` to discuss the ongoing project developments, updates, and plans, share use cases and stories or ask/answer questions about BitOps tool.

See https://github.com/bitovi/bitops/discussions/288 for the meeting link and schedule.

### Follow-ups from the previous #366 Meeting
- BitOps landing page (@mickmcgrath13 @armab)
  - BitOps 1-3min intro Video (@PhillypHenning)
- Update [BitOps HeyEmoji blog](https://www.bitovi.com/blog/heyemoji-slack-app-aws-terraform-ansible) to v2 (@armab)
- Update [Combine Terraform and Ansible to Provision and Configure a Web Server](https://www.bitovi.com/blog/bitops-terraform-ansible) to v2 (@LeoDiazL)
- BitOps Twitch Try

### BitOps in Github Actions
Current state, examples, and progress of using BitOps in GH Actions.

### Discussion: BitOps CLI Repo Generator - 3 CLI Options
Are we ready to make a decision about the future of the BitOps Ops Repo CLI generator?
@armab to also highlight Yoman CLI UX: what's good about the existing https://github.com/bitovi/bitops-operations-repo-generator

### v2.3.0 Release Planning
- https://github.com/bitovi/bitops/milestone/9
- Before the end of December
- Release Announcement blog (v2.1, v2.2, v2.3 highlights)
 
### Development: Local plugin install (@mickmcgrath13)
@mickmcgrath13 to highlight https://github.com/bitovi/bitops/pull/349 new feature which allows setting local `file://` for plugin install instead of remote git clone.

**Comment by mickmcgrath13** [comment]
# Call Notes

## Follow-ups from the previous https://github.com/bitovi/bitops/discussions/366 Meeting
- BitOps landing page (@mickmcgrath13 @armab)
- BitOps 1-3min intro Video (@PhillypHenning)
- Update [BitOps HeyEmoji blog](https://www.bitovi.com/blog/heyemoji-slack-app-aws-terraform-ansible) to v2 (@armab)
  - waiting on a PR from mmcdole/heyemoji
- Update [Combine Terraform and Ansible to Provision and Configure a Web Server](https://www.bitovi.com/blog/bitops-terraform-ansible) to v2 (@LeoDiazL)
  - includes "last updated" snippet
- BitOps Twitch Try
  - First twitch scheduled for our DevOps holiday 2022

## BitOps in Github Actions
- Github Action: https://github.com/bitovi/github-actions-deploy-docker-to-ec2

## Development: Local plugin install
- https://github.com/bitovi/bitops/pull/349

## BitOps CLI Repo Generator - 3 CLI Options
- https://github.com/bitovi/bitops/discussions/353
- Go with Go for now! 🎉 
- maybe also write DeCLI in go, too
- Next steps: need a github issue for a go-based CLI generator written in GO (cc @armab )

## v2.3.0 Release Planning
- metrics dashboard
- ideally would get local plugin install for this release (https://github.com/bitovi/bitops/pull/349)

**Comment by arm4b** [comment]
## Meeting Notes

### Follow-ups from the previous #366 Meeting

- BitOps landing page (@mickmcgrath13 @armab)
  - no progress on that, TBD
- BitOps 1-3min intro Video
  -  WIP by @PhillypHenning with re-recorded new videos based on feedback
- Update [BitOps HeyEmoji blog](https://www.bitovi.com/blog/heyemoji-slack-app-aws-terraform-ansible) to v2 (@armab)
  - Docs RFR: 
  - Needs publishing
  - A great way to play with BitOps, find bugs, and enhance the tool
- [Combine Terraform and Ansible to Provision and Configure a Web Server](https://www.bitovi.com/blog/bitops-terraform-ansible) to v2
  - Done! 🎉  by @LeoDiazL 
  - Required Yoman generator npm release, bugfixes, bitops release, blog publish access
- Phil: attending a https://www.techto.org/ conference in Toronto
  - might highlight BitOps there
- Twitch:
   - First BitOps Twitch in December
   - Needs an agenda for the next year BitOps 
     - there are lots of opportunities and ideas
   - @PhillypHenning is owning the Meetings & Twitch

### BitOps in Github Actions (overview by @mickmcgrath13)
- Enters BitOps as a Platform Engineering middleware
- Via GH Actions it's easy to use
    - less technical
      - Exactly our target group for BitOps! 🎉 
    - less devops-y
    - under the hood using BitOps
    - GH Actions marketplace
    - Action inputs are easy to configure
- Actions:
    - StackStorm Action - WIP
    - Heyemoji Action
      - https://github.com/marketplace/actions/deploy-docker-to-aws-ec2
- Opens a lot of opportunities

### Discussion: BitOps CLI Repo Generator - 3 CLI Options
#### Yoman CLI UX
What's good about the existing https://github.com/bitovi/bitops-operations-repo-generator:
- Colors
- Yes/No/Defaults inputs
- Easy to use, nice output
- Shows generated structure
- Shows generated structure diff on the 2nd run
- Generates ready-to-use examples based on templates (https://github.com/bitovi/bitops-operations-repo-generator/tree/master/generators/app/templates) for each plugin
- The code is very easy, `100 lines`: https://github.com/bitovi/bitops-operations-repo-generator/blob/master/generators/app/index.js
- We want a similar experience with the new tool (not worse)
  - but using the alternative installation method (not `npm`)
  - Gotcha: with `Golang`, will require more lower-level code

#### Decision on BitOps CLI
- pip cli is OK, Docker CLI is easiest, Golang is cool, npm is not a good fit
- Mick: Golang would be a good opportunity to get more 👍 from the community
  - is great considering the binary portability
  - **Decided: go with Go** based on the previous conversations and opinions
- Eugen: a language the Cloud-Native community speaks. Go in the wishlist too, but will require lower-level code and more dev overhead.
- Phil: is super excited about playing with Go
- Karam: will be interested to write Golang too, a language on her wishlist
- TODO:
  - we need a dedicated repo for `bitops-cli`
  - CI/CD, release, build pipelines
  - future work: integrate with `brew` for an easier install
- Future opportunity:
  - DeCLI tool https://github.com/bitovi/bitops/issues/230

### Development: Local plugin install (@mickmcgrath13)
@mickmcgrath13 highlighted https://github.com/bitovi/bitops/pull/349 new feature which allows setting local `file://` for plugin install instead of remote git clone.
- We want it for the v2.3.0 release
- Need a few more cycles of updates to address review comments
- Implements https://github.com/bitovi/bitops/issues/308
- Helps with https://github.com/bitovi/bitops/issues/224
- Helps a lot with the future CI for the Plugins:
  - so `plugin install` is tested on every PR when someone creates a change in the plugin's code


### v2.3.0 Release Planning
- https://github.com/bitovi/bitops/milestone/9
  - We want it before the end of December
  - Release Announcement blog (v2.1, v2.2, v2.3 highlights)
- Mick highlights the need for the OSS Metrics Dashboard
- Update: 
  - next BitOps meeting ~21th Dec
  - Phil: majority of the team will be off already
  - realistically we have 1 week to cut it
    - include a few more fixes
    - and write a blog post
- Re-evaluate the [milestone plan](https://github.com/bitovi/bitops/milestone/9) & features:
  - Definitely 2.3:
    - https://github.com/bitovi/bitops/pull/349
    - https://github.com/bitovi/bitops/issues/374
    - https://github.com/bitovi/bitops/issues/358
    - https://github.com/bitovi/bitops/pull/368
  - Maybe 2.3:
    - https://github.com/bitovi/bitops/issues/357
    - https://github.com/bitovi/bitops/issues/360
    - https://github.com/bitovi/bitops/issues/286
    - https://github.com/bitovi/bitops/issues/329
    - https://github.com/bitovi/bitops/issues/371
- Conclusion:
  - Considering 1+ week left, include what's possible (ideally as many bugfixes as possible)
  - Cut a release
  - Write a blog post
  - Re-assign non-implemented things to the next 2.4 release
